analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/ConditionEvaluatesToConstantBase.cs

```diff
@@ -93,61 +91,66 @@ public override void ExecutionCompleted()
     private void ReportIssue(IOperation operation, BasicBlock block, bool conditionValue)
     {
         var issueMessage = operation.Kind == OperationKindEx.IsNull ? MessageNull : string.Format(MessageBool, conditionValue);
-        var secondaryLocations = SecondaryLocations(block, conditionValue);
+        var syntax = ToBranchValueCondition(operation.Syntax);
+        var secondaryLocations = SecondaryLocations(block, conditionValue, syntax);
         if (secondaryLocations.Any())
         {
-            ReportIssue(Rule2583, operation, secondaryLocations, issueMessage, S2583MessageSuffix);
+            ReportIssue(Rule2583, syntax, secondaryLocations, issueMessage, S2583MessageSuffix);
         }
         else
         {
-            ReportIssue(Rule2589, operation, null, issueMessage, string.Empty);
+            ReportIssue(Rule2589, syntax, null, issueMessage, string.Empty);
         }
     }
 
-    private List<Location> SecondaryLocations(BasicBlock block, bool conditionValue)
+    private List<Location> SecondaryLocations(BasicBlock block, bool conditionValue, SyntaxNode conditionSyntax)
     {
         List<Location> locations = new();
-        IOperation currentStart = null;
-        IOperation currentEnd = null;
-        var unreachable = UnreachableOperations(block, conditionValue).Where(x => !IsIgnoredLocation(x.Syntax)).ToHashSet();
-        foreach (var operation in unreachable.Concat(reached).OrderBy(x => x.Syntax.SpanStart))
+        var unreachable = UnreachableOperations(block, conditionValue);
+        var currentStart = conditionSyntax.Span.End;
+        var reachedNodes = reached.Select(x => x.Syntax).Where(x => x.SpanStart > conditionSyntax.Span.End).OrderBy(x => x.SpanStart);
+
+        foreach (var reachedNode in reachedNodes)
         {
-            if (unreachable.Contains(operation))
-            {
-                currentStart ??= operation;
-                currentEnd = operation;
-            }
-            else
+            if (AddLocation(reachedNode.SpanStart))
             {
-                AddCurrent();
+                currentStart = reachedNode.Span.End;
             }
         }
-        AddCurrent();
+        AddLocation(int.MaxValue);
```

Comment:
Why here you are setting as end the `int.MaxValue`?

Reply:
To find unreachable operations that are after the last reached operation.